### PR TITLE
Publish to AWS serverless repository

### PR DESCRIPTION
**Pull request by zaro0508**
* Setup to publish to AWS serverless repository
* Update documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,10 +61,32 @@ sam package --template-file .aws-sam/build/template.yaml \
...
 ```
 
+## Publish Lambda
+
+### Private access
+Publishing the lambda makes it available in your AWS account.  It is accessible in
+the [serverless application repository](https://console.aws.amazon.com/serverlessrepo).
+
+```shell script
+sam publish --template .aws-sam/build/lambda-template.yaml
```
Comment (cascadianblue) [inline README.md:71]: I believe a user of this template would have to replace `lambda-template.yaml` with the name of their own repository, correct? I don't think that's clear from this README

File: README.md [hunk 2]
```diff
@@ -83,6 +104,13 @@ Install the lambda using sceptre:
 sceptre --var "profile=my-profile" --var "region=us-east-1" launch prod/lambda-template.yaml
```
Comment (cascadianblue) [inline README.md:104]: "curl https://s3.amazonaws.com/bootstrap-awss3cloudformationbucket-19qromfd235z9/lambda-template/master/lambda-template.yaml"

Might as well switch to [new-style S3 URLs](https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/) while you are doing a PR

File: README.md (hunk 1 above)
Comment (zaro0508) [reply README.md:71]: not really, it's the template file specified with the `--output-template-file` flag in the  previous `sam package` command.  We have designed our deploy strategy to have matching repo name and output templates.

We can expect lambda devs to have a basic understanding of the SAM framework and realize that.

File: README.md [hunk 3]
```diff
@@ -62,10 +61,32 @@ sam package --template-file .aws-sam/build/template.yaml \
 aws s3 cp .aws-sam/build/lambda-template.yaml s3://bootstrap-awss3cloudformationbucket-19qromfd235z9/lambda-template/master/
 ```
 
+## Publish Lambda
+
+### Private access
+Publishing the lambda makes it available in your AWS account.  It is accessible in
```
[the remark this answers is not in the document]
Comment (zaro0508) [reply README.md]: fixed

File: README.md [hunk 4]
```diff
@@ -1,6 +1,5 @@
 # lambda-template
-Template for creating lambda repositories
-A template for quickly starting a new AWS lambda project.
+A github template for quickly starting a new AWS lambda project.
```
[the remark this answers is not in the document]
Comment (zaro0508) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (zaro0508) [reply README.md:104]: fixed

**Review by cascadianblue — APPROVED** [review]

**Pull request merged**